edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Core.Test/config/ConfigUpdaterTest.cs

```diff
@@ -89,10 +89,15 @@ public async Task TestPeriodicConfigUpdate()
 
             // Act
             var configUpdater = new ConfigUpdater(router, messageStore.Object, updateFrequency);
-            configUpdater.Init(configProvider.Object);
+            await configUpdater.Init(configProvider.Object);
 
             // Assert
-            await Task.Delay(TimeSpan.FromSeconds(8));
+            configProvider.Verify(c => c.GetConfig(), Times.Once);
+            endpointExecutorFactory.Verify(e => e.CreateAsync(It.IsAny<Endpoint>()), Times.Once);
+            messageStore.Verify(m => m.SetTimeToLive(It.IsAny<TimeSpan>()), Times.Once);
+
+            // After 5 seconds, the periodic task should not have run.
```

Comment:
`TestPeriodicConfigUpdate` runs `Init`, verifies the config, then waits 5 seconds and verifies it again. In light of that logic, I don't understand this comment:
```
// After 5 seconds, the periodic task should not have run.
```

If the periodic task _did_ run during that 5-second sleep, would the config somehow be different the second time you check it?

Reply:
Ah, nevermind. These Verify statements aren't checking the contents of the config, they're checking that certain functions on the mock objects were called exactly once. So this confirms that the periodic task hasn't run yet. 👍

Reply:
Right, it just makes sure that we are not pulling the config too often